### ISSUE TITLE
Consider converting waitTimeForCompletedObjectiveIds to pub/sub

### ISSUE THREAD

**Issue by kerzhner** [statechannels/go-nitro#1119]
At the moment, [waitTimeForCompletedObjectiveIds](https://github.com/statechannels/go-nitro/blob/d31553177acba50693d458da9a3ef2774af30c29/client_test/helpers_test.go#L31) uses the [CompletedObjectives](https://github.com/statechannels/go-nitro/blob/d31553177acba50693d458da9a3ef2774af30c29/client/client.go#L92) channel listen for completed objectives. If more than one consumer calls  `waitTimeForCompletedObjectiveIds` with the same objective id, then only one consumer will be unblocked when the objective completes. This is the result of using golang channel producer/consumer pattern. The rest of consumers will block forever.

We should consider modifying `waitTimeForCompletedObjectiveIds` into a publish/subscribe implementation. This way consumers can listen for the objective completed notification for same objective id. We could even modify `completedObjectives` channel into a  publish/subscribe implementation.

One note is to be aware of race conditions. Golang channel producer/consumer pattern very nicely avoids race conditions. With publisher/subscriber model, a subscriber will need to start listening for notifications before a publisher posts a notification.

**Comment by kerzhner** [comment]
This issue is a duplicate of https://github.com/statechannels/go-nitro/issues/1114.
